clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -2327,38 +2340,36 @@ private synchronized VeniceWriter<KafkaKey, byte[], byte[]> getVeniceWriter(Push
 
   private synchronized Properties getVeniceWriterProperties(PushJobSetting pushJobSetting) {
     if (veniceWriterProperties == null) {
-      veniceWriterProperties = createVeniceWriterProperties(pushJobSetting.kafkaUrl, pushJobSetting.sslToKafka);
+      veniceWriterProperties = createVeniceWriterProperties(pushJobSetting);
     }
     return veniceWriterProperties;
   }
 
-  private Properties createVeniceWriterProperties(String kafkaUrl, boolean sslToKafka) {
+  private Properties createVeniceWriterProperties(PushJobSetting pushJobSetting) {
     Properties veniceWriterProperties = new Properties();
-    veniceWriterProperties.put(KAFKA_BOOTSTRAP_SERVERS, kafkaUrl);
+    veniceWriterProperties.put(KAFKA_BOOTSTRAP_SERVERS, pushJobSetting.kafkaUrl);
     veniceWriterProperties.put(VeniceWriter.MAX_ELAPSED_TIME_FOR_SEGMENT_IN_MS, -1);
     if (props.containsKey(VeniceWriter.CLOSE_TIMEOUT_MS)) { /* Writer uses default if not specified */
       veniceWriterProperties.put(VeniceWriter.CLOSE_TIMEOUT_MS, props.getInt(VeniceWriter.CLOSE_TIMEOUT_MS));
     }
-    if (sslToKafka) {
-      veniceWriterProperties.putAll(this.sslProperties.get());
-    }
-    if (props.containsKey(KAFKA_PRODUCER_REQUEST_TIMEOUT_MS)) {
+    if (props.containsKey(VeniceWriter.LARGE_RECORDS_ALLOWED)) {
       veniceWriterProperties
-          .setProperty(KAFKA_PRODUCER_REQUEST_TIMEOUT_MS, props.getString(KAFKA_PRODUCER_REQUEST_TIMEOUT_MS));
-    } else {
-      veniceWriterProperties.setProperty(KAFKA_PRODUCER_REQUEST_TIMEOUT_MS, Integer.toString(Integer.MAX_VALUE));
-    }
-    if (props.containsKey(KAFKA_PRODUCER_RETRIES_CONFIG)) {
-      veniceWriterProperties.setProperty(KAFKA_PRODUCER_RETRIES_CONFIG, props.getString(KAFKA_PRODUCER_RETRIES_CONFIG));
-    } else {
-      veniceWriterProperties.setProperty(KAFKA_PRODUCER_RETRIES_CONFIG, Integer.toString(Integer.MAX_VALUE));
-    }
-    if (props.containsKey(KAFKA_PRODUCER_DELIVERY_TIMEOUT_MS)) {
-      veniceWriterProperties
-          .setProperty(KAFKA_PRODUCER_DELIVERY_TIMEOUT_MS, props.getString(KAFKA_PRODUCER_DELIVERY_TIMEOUT_MS));
-    } else {
-      veniceWriterProperties.setProperty(KAFKA_PRODUCER_DELIVERY_TIMEOUT_MS, Integer.toString(Integer.MAX_VALUE));
-    }
+          .put(VeniceWriter.LARGE_RECORDS_ALLOWED, props.getBoolean(VeniceWriter.LARGE_RECORDS_ALLOWED));
+    }
+    // TODO: difference between venicewriter and non-vw properties?
+    veniceWriterProperties.put(VeniceWriter.MAX_RECORD_SIZE_BYTES, pushJobSetting.maxRecordSizeBytes);
```

Comment:
Can we collapse these two configs into one?
When `VeniceWriter.MAX_RECORD_SIZE_BYTES` is `-1`, VeniceWriter would treat the max limit as unlimited.

Reply:
As you and @nisargthakkar noticed, it's not needed in the context of just this PR and enforcing the limit on Batch Push Jobs.

This feature flag is to protect against `VeniceWriter` unintentionally enforcing limits on nearline jobs, which will be handled in a later PR.

1. The conversion of the unset value (`-1`) into the default limit (currently `10MB`) happens in the `VeniceWriter`, because both the batch and nearline job code paths will go through it, and that's also where the check happens ([link](https://github.com/linkedin/venice/blob/4ba73c94710e8fad78e4f82c58f70a2b1a5b8bfc/internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java#L896)).
2. It also comes with the benefit of having a feature flag so we can easily turn off and on the feature if we need to do so in the future.

The other way to do it would be to set the default limit individually inside the `VenicePushJob` and `Consumer`, but I felt that the feature flag would be a clean approach that has added benefits.


As mentioned in your original design doc:

> Venice will have a default max allowed record size when chunking is enabled, such as 5MB (for example).

I noticed that there is already a `~1MB` limit ([`maxSizeForUserPayloadPerMessageInBytes`](https://github.com/linkedin/venice/blob/4ba73c94710e8fad78e4f82c58f70a2b1a5b8bfc/internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java#L896)) to handle the Kafka size limit, and if chunking is not enabled then an exception will be thrown. So if there are records that need to be chunked, chunking should be enabled and the default max allowed record size should always take effect after the `~1MB` limit.

Reply:
It is fine if it can make the logic easier, but I don't like to hard-code the max limit in `VeniceWriter` and it needs to be configurable as we may change it and apply it globally for the stores, whose store config value is `-1`.

Reply:
With that approach, I will remove the feature flag and the two code path locations (`VenicePushJob` for batch and `Consumer` for nearline) will read the default value from the controller and that's how the limit will be enforced.